pkg/tasks/kubestatemetrics.go

```diff
@@ -95,6 +95,21 @@ func (t *KubeStateMetricsTask) Run(ctx context.Context) error {
 		return fmt.Errorf("reconciling kube-state-metrics Deployment failed: %w", err)
 	}
 
+	prometheusK8sTokenSecret, err := t.client.GetTokenSecret(ctx, dep.Namespace, "prometheus-k8s")
```

Comment:
I've used the `prometheus-k8s-token-...` secret to query the `/metrics` endpoint, since that was [allowed by KRP](https://github.com/openshift/cluster-monitoring-operator/blob/6795f509e77cce6d24e5a3e371a432ca22e1a8e7/assets/kube-state-metrics/kube-rbac-proxy-secret.yaml#L17).